src/common/com/intellij/plugins/haxe/ide/HaxeLookupElement.java

```diff
@@ -107,6 +112,28 @@ public void renderElement(LookupElementPresentation presentation) {
     }
   }
 
+  @Override
+  public void handleInsert(InsertionContext context) {
+    HaxeMemberModel memberModel = HaxeMemberModel.fromPsi(myComponentName);
+    boolean hasParams = false;
+    boolean isMethod = false;
+    if (memberModel != null) {
+      if (memberModel instanceof HaxeMethodModel) {
+        isMethod = true;
+        HaxeMethodModel methodModel = (HaxeMethodModel)memberModel;
+        hasParams = !methodModel.getParametersWithContext(this.context).isEmpty();
+      }
+    }
+
+    if (isMethod) {
+      final LookupElement[] allItems = context.getElements();
+      final boolean overloadsMatter = allItems.length == 1 && getUserData(FORCE_SHOW_SIGNATURE_ATTR) == null;
+      JavaCompletionUtil.insertParentheses(context, this, overloadsMatter, hasParams);
+    }
+  }
+  // Workaround for IDEA 14. There must be JavaCompletionUtil.FORCE_SHOW_SIGNATURE_ATTR
+  private static final Key<Boolean> FORCE_SHOW_SIGNATURE_ATTR = Key.create("forceShowSignature");
```

Comment:
Ok, I chose to go with reflections. In this case the attribute `JavaCompletionUtil.FORCE_SHOW_SIGNATURE_ATTR` may be renamed in future and we will got runtime exception here with no compiler errors. Hope IDEA developers will not do this.
The advantage using reflections is that we have all the code in one place.